src/tactic/rename.lean

```diff
@@ -1,119 +0,0 @@
-`unfreeze_local_instances`.
-
-If `strict` is true, we fail if `name_map` refers to hypotheses that do not
-appear in the local context or that appear before a frozen local instance.
-Conversely, if `strict` is false, some entries of `name_map` may be silently
-ignored.
-
-Note that we allow shadowing, so renamed hypotheses may have the same name
-as other hypotheses in the context.
--/
-meta def rename' (renames : name_map name) (strict := tt) : tactic unit := do
-  ctx ← revertible_local_context,
-  when strict (do
-    let ctx_names := rb_map.set_of_list (ctx.map expr.local_pp_name),
-    let invalid_renames :=
-      (renames.to_list.map prod.fst).filter (λ h, ¬ ctx_names.contains h),
-    when ¬ invalid_renames.empty $ fail $ format.join
-      [ "Cannot rename these hypotheses:\n"
-      , format.intercalate ", " (invalid_renames.map to_fmt)
-      , format.line
-      , "This is because these hypotheses either do not occur in the\n"
-      , "context or they occur before a frozen local instance.\n"
-      , "In the latter case, try `tactic.unfreeze_local_instances`."
-      ]),
-  let new_name := λ current_name,
-    (renames.find current_name).get_or_else current_name,
-  let intro_names := list.map (new_name ∘ expr.local_pp_name) ctx,
-  revert_lst ctx,
-  intro_lst intro_names,
-  pure ()
-
-end tactic
-
-
-namespace tactic.interactive
-
-/-- Parse a current name and new name for `rename'`. -/
-meta def rename'_arg_parser : parser (name × name) :=
-  prod.mk <$> ident <*> (optional (tk "->") *> ident)
-
-/-- Parse the arguments of `rename'`. -/
-meta def rename'_args_parser : parser (list (name × name)) :=
-  (functor.map (λ x, [x]) rename'_arg_parser)
-  <|>
-  (tk "[" *> sep_by (tk ",") rename'_arg_parser <* tk "]")
-
-/--
-Rename one or more local hypotheses. The renamings are given as follows:
-
-```
-rename' x y             -- rename x to y
-rename' x → y           -- ditto
-rename' [x y, a b]      -- rename x to y and a to b
-rename' [x → y, a → b]  -- ditto
-```
-
-Brackets are necessary if multiple hypotheses should be renamed in parallel.
--/
-meta def rename' (renames : parse rename'_args_parser) : tactic unit :=
-  tactic.rename' (rb_map.of_list renames)
-
-/--
-Renames one or more hypotheses in the context.
-
-```lean
-example {α β} (a : α) (b : β) : unit :=
-begin
-  rename' a a',              -- result: a' : α, b  : β
-  rename' a' → a,            --         a  : α, b  : β
-  rename' [a a', b b'],      --         a' : α, b' : β
-  rename' [a' → a, b' → b],  --         a  : α, b  : β
-  exact ()
-end
-```
-
-Compared to the standard `rename` tactic, this tactic makes the following
-improvements:
-
-- You can rename multiple hypotheses at once.
-- Renaming a hypothesis always preserves its location in the context (whereas
-  `rename` may reorder hypotheses).
--/
-add_tactic_doc
-{ name       := "rename'",
-  category   := doc_category.tactic,
-  decl_names := [`tactic.interactive.rename'],
-  tags       := ["renaming"] }
```

Comment:
This should be added back to https://github.com/leanprover-community/mathlib/blob/master/src/tactic/lean_core_docs.lean

Reply:
`lean_core_docs` already has this entry:

```lean
add_tactic_doc
{ name       := "rename",
  category   := doc_category.tactic,
  decl_names := [`tactic.interactive.rename],
  tags       := ["core", "renaming"] }
```
That should be enough, shouldn't it?

Reply:
Oh, I missed that! Sorry for the noise.